src/main/java/org/swisspush/redisques/RedisQues.java

```diff
@@ -121,27 +122,32 @@ private void handleRegistrationRequest(Message<String> msg) {
             log.debug("RedisQues Got registration request for queue " + queueName + " from consumer: " + uid);
         }
         // Try to register for this queue
-        SetOptions setOptions = new SetOptions().setNX(true).setEX(consumerLockTime);
-        redisClient.setWithOptions(consumersPrefix + queueName, uid, setOptions, event -> {
+        redisAPI.setnx(consumersPrefix + queueName, uid, event -> {
```

Comment:
Thx @mcweba, I think you have a point here, we need to make it atomic.

Can you confirm that the basic working principle is
---
Add (set) it if not already there, otherwise don't touch (e.g. do not update the expiry).
---
And we do need it to be exactly like this, correct?

Reply:
@dominik-cnx Yes exactly. It is crucial that an existing entry (of another consumer of this queue) is not overwritten.

Reply:
@mcweba kindly review the revised implementation, thx.